src/Components/Feature/TodoItemDetail.tsx

```diff
@@ -0,0 +1,45 @@
+import { useState, useEffect } from "react";
+import { IToDoResp } from "../../Shared/types";
+import useFetchById from "../Common/useFetchById";
+import Todo from "./ToDo";
+import { useParams } from "react-router-dom";
+
+const TodoItemDetail = () =>{
+    const { id } = useParams();
+
+    const [todos, setTodos] = useState<IToDoResp | null>(null);
+
+    const { data, loading, error }:{ data:IToDoResp | null; loading: boolean; error: string | null } = useFetchById('https://jsonplaceholder.typicode.com/todos/'+id);
+  
+    useEffect(() => {
+      if (data) {
+        setTodos(data);
+      }
+    }, [data]);
+  
+    const markCompleted = (id: number) => {
```

Comment:
all the handlers is repeated in this file please create reusable handlers in helper file

Reply:
I have removed where its do not needed